src/Protobuf/Types/Int64.elm

```diff
@@ -0,0 +1,60 @@
+module Protobuf.Types.Int64 exposing
+    ( Int64, Int32s
+    , fromInt32s, toInt32s
+    )
+
+{-| A simple 64-bit integer made up from two 32-bit integers.
+This is used by the `int64` and related en/decoders.
+This module is intentionally kept sparse to achieve the following goals:
+
+  - Support Int64 data types in elm-protocol-buffers
+  - Don't add extra code or dependencies not required for the main goal of this package (serialization and deserialization)
+
+If you find yourself needing extra logic/arithmetic revolving around 64-bit integers,
+please use one of the available packages providing what you need and convert with
+`fromInt32s` and `toInt32s` between the two data types.
+
+
+# Data Types
+
+@docs Int64, Int32s
+
+
+# Conversions
+
+@docs fromInt32s, toInt32s
+
+-}
+
+import Bitwise
+
+
+{-| The `Int64` data type. Guarantees the invariant that the internal integers are kept between `-2 ^ 31` and `2 ^ 31 - 1`.
+-}
+type Int64
+    = Int64 Int32s
+
+
+{-| A record containing two `Int` values, one for the lower 32 bits and one for the upper 32 bits.
+-}
+type alias Int32s =
+    { lower : Int, upper : Int }
+
+
+{-| Build an `Int64` from two `Int` values.
+-}
+fromInt32s : Int32s -> Int64
+fromInt32s =
+    rewrap >> Int64
+
+
+{-| Get the two `Int` values for lower and upper bits from an `Int64`.
+-}
+toInt32s : Int64 -> Int32s
```

Comment:
This actually needs to be exposed. Otherwise there is no way for users of the library to "replace" the Int64 type defined here with a type of their choice.

Reply:
For some reason I cannot respond to your other comment, so I'll comment here.

If I'm understanding correctly you want `fromInt32` to have a signature of `Int -> Int -> Int64`.
I wouldn't mind to change it, but I do think that the order of arguments is a bit ambiguous.
Both ``lower -> higher -> Int64` and `higher -> lower -> Int64` kind of make sense.
If you don't think so, please tell me which one to implement ;)

Reply:
* If users can see the internal structure (alias) of the `Int64` they can just take out whatever they need, right? Shouldn't matter if they use the alias or an int array;
* I agree it is a bit ambiguous, but as you can see in your test it is also easy to mix the order of the record keys, which is likewise confusing imho. I would to `higher -> lower -> Int64` as that's how we write normal numbers as well.

Reply:
- Right now they cannot see the internal structure since the `Int64` constructor is not exposed, just the opaque Int64 type.
This is important imo, because the rest of the serialization code relies on the fact that the lower and higher bits are between -2^31 and 2^31 - 1.

- Okay, I'll change the function names to `fromInts` and `toInts` and use `Int -> Int -> Int64` as the type signature for the former (with `higher` as the first argument)

Reply:
* Ah, I see. I missed why you needed the `Int64` constructor, but I agree that is needed;
* Can we then have `toInts` return `(Int, Int)` rather than the internal alias. I would not expose that one as it is our internal representation.

Reply:
Sure, I moved the constructor and the functions to Internal.Int64, now Protobuf.Types.Int64 just exposes aliases for
`Int64`, `toInts` and `fromInts`.